src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.Stream.cs

```diff
@@ -0,0 +1,347 @@
+        /// </summary>
+        private sealed class ODataUtf8JsonWriteStream : Stream
+        {
+            private readonly ODataUtf8JsonWriter jsonWriter = null;
+            private byte[] buffer;
+            int numBytesNotWrittenFromPreviousChunk = 0;
+
+            /// <summary>
+            /// Initializes a new instance of the ODataUtf8JsonWriterStream class with the specified ODataUtf8JsonWriter.
+            /// </summary>
+            /// <param name="writer">The OData UTF-8 JSON writer to write to.</param>
+            internal ODataUtf8JsonWriteStream(ODataUtf8JsonWriter writer)
+            {
+                this.jsonWriter = writer;
+            }
+
+            public override bool CanRead => false;
+
+            public override bool CanSeek => false;
+
+            public override bool CanWrite => true;
+
+            /// <summary>
+            /// Gets the length in bytes of the stream. This is not supported by this stream.
+            public override long Length => throw new NotSupportedException();
+
+            /// <summary>
+            /// Gets or sets the position within the stream. This is not supported by this stream.
+            /// </summary>
+            public override long Position { get => throw new NotSupportedException(); set => throw new NotSupportedException(); }
+
+            /// <summary>
+            /// Flushes any buffered data to the underlying stream synchronously.
+            /// </summary>
+            public override void Flush()
+            {
+                this.jsonWriter.Flush();
+            }
+
+            /// <summary>
+            /// Flushes any buffered data to the underlying stream asynchronously.
+            /// </summary>
+            /// <param name="cancellationToken">A A cancellation token to observe while waiting for the flush operation to complete.</param>
+            /// <returns>A task representing the asynchronous flush operation.</returns>
+            public override async Task FlushAsync(CancellationToken cancellationToken)
+            {
+                await this.jsonWriter.FlushAsync().ConfigureAwait(false);
+            }
+
+            /// <summary>
+            /// Disposes the object.
+            /// </summary>
+            /// <param name="disposing">True if called from Dispose; false if called form the finalizer.</param>
+            protected override void Dispose(bool disposing)
+            {
+                if (this.numBytesNotWrittenFromPreviousChunk > 0)
+                {
+                    // If there are unprocessed bytes, encode and write them as the final block.
+                    ReadOnlySpan<byte> bytesNotProcessedFromPreviousChunk = this.buffer.AsSpan().Slice(0, this.numBytesNotWrittenFromPreviousChunk);
+
+                    this.jsonWriter.Base64EncodeAndWriteChunk(bytesNotProcessedFromPreviousChunk, isFinalBlock: true, out numBytesNotWrittenFromPreviousChunk);

```

Comment:
@ElizabethOkerio Curious about the logic here. `this.numBytesNotWrittenFromPreviousChunk` is an out parameter here and in the statement that follows, we set it's value to 0. Could you explain the reasoning behind that logic?

Reply:
If we don't care about the value that is returned, maybe we should use `out _`. Or if we're sure in this scenario out parameter will be assigned a value of 0, we should remove the next line that assigns a value of 0 to `numBytesNotWrittenFromPreviousChunk`.
In addition, if we expect it to be 0 but we'd like to be sure, we could add a check and throw an exception if it's not.

Reply:
This is when we are dispoing.. At this point this is the final block that's being written and we do not expect the `numBytesNotWrittenFromPreviousChunk` to have anything. That's why it's set to zero so that we start with 0 unwritten bytes in the next write.

Reply:
@ElizabethOkerio Explain "we don't expect the `numBytesNotWrittenFromPreviousChunk` to have anything"? This is a non-nullable out parameter and the value will be set from `Base64EncodeAndWriteChunk`.
In the case when we're disposing, will `Base64EncodeAndWriteChunk` have set it's value to 0? If the answer to that is yes, then we shouldn't have a redundant statement that set's it to 0 again. If the answer is no, then I'd be concerned because the out parameter should return the number of bytes not written from previous chunk...